%&latex
\documentclass[12pt]{article}
\usepackage{amsthm,amsmath}
\usepackage{graphicx,psfrag,epsf}
\usepackage{enumerate}
\usepackage{natbib}
\usepackage{url} % not crucial - just used below for the URL
\usepackage{amsthm,amsmath}
\usepackage[utf8]{inputenc}

%\pdfminorversion=4
% NOTE: To produce blinded version, replace "0" with "1" below.
\newcommand{\blind}{0}

% DON'T change margins - should be 1 inch all around.
\addtolength{\oddsidemargin}{-.5in}%
\addtolength{\evensidemargin}{-.5in}%
\addtolength{\textwidth}{1in}%
\addtolength{\textheight}{1.3in}%
\addtolength{\topmargin}{-.8in}%


\begin{document}

%\bibliographystyle{natbib}
%\bibliographystyle{bmc-mathphys}
\bibliographystyle{agsm}

\def\spacingset#1{\renewcommand{\baselinestretch}%
{#1}\small\normalsize} \spacingset{1}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\if0\blind
{
  \title{\bf Promoting analysis reproducibility with accessibility: An example in evolutionary biology}
  \author{Luna L. Sanchez Reyes\thanks{
    The authors gratefully acknowledge ``Sustaining the Open Tree of Life``, NSF ABI No. 1759838, and ABI No. 1759846.}\hspace{.2cm}\\
    School of Natural Sciences, University of California, Merced\\
    and \\
    Emily Jane McTavish \\
    School of Natural Sciences, University of California, Merced}
  \maketitle
} \fi

\if1\blind
{
  \bigskip
  \bigskip
  \bigskip
  \begin{center}
    {\LARGE\bf Promoting analysis reproducibility with accessibility: An example in evolutionary biology}
\end{center}
  \medskip
} \fi

\bigskip
\begin{abstract}
Reproducibility is essential for scientific development. Efforts
to increase scientific reproducibility have focused on increasing availability
of code and data. However, availability does not imply accessibility, and the latter
is infrequently addressed, even if it is key to achieve full workflow automatization
and reproducibility. Using an example in evolutionary biology, we identify factors
that have specifically affected accessibility in
the natural sciences, and ways researchers can address this to ensure reproducible
and automatic workflows.
The Open Tree of Life project (OpenTree) has developed a platform that facilitates
availability of results from evolutionary biology research. However, baseline
computational knowledge and skills required to
access scientific results are often not found among target users. While documentation
is available, it is often written using highly specialized language that is also
 inaccessible for the average target user.
We identified a set of principles to improve accessibility of OpenTree scientific data,
implement them in a tutorial, and elaborate on their general application to code and
documentation of scientific workflows.



\end{abstract}

\noindent%
{\it Keywords:}  open science, education, R, phylogenetics, tutorials
\vfill

\newpage
\spacingset{1.45} % DON'T change the spacing!
\section*{Introduction}
\label{sec:intro}

Reproducibility --the extent to which consistent results are obtained when a scientific
research experiment is repeated \citep{repdef2021}-- is a key aspect for the advancement
of science, as it constitutes a minimum standard that allows understanding research products
(e.g., methods, data, analysis, results, etc.),
to determine their reliability and generality, and eventually build up scientific
knowledge and applications based on those products
\citep{king1995replication, peng2011reproducible, powers2019open}.
In the natural sciences, rates of reproducibility are low \citep{ioannidis2005most, prinz2011believe},
which has elicited concerns about a reproducibility crisis in the field \citep{baker2016reproducibility}.
%%% The scientific community has united to incentivize cultural changes that will improve
%%% reproducibility rates long term, such as transparency, availability, and workflow
%%% automatization, to name a few \citep{peng2015reproducibility}.
In response, the scientific community has been developing new principles and standards to incentivize
cultural changes in an effort to improve long term reproducibility rates in the natural sciences
\citep{peng2015reproducibility, wilkinson2016fair}.
A standard for reproducibility that has received much attention is scientific availability, which
we define as the property of research products to be acquired, copied, analyzed,
processed and/or reused, at no financial, legal or technical cost \citep{arnold2019turing},
and with no geographic, demographic or temporal barriers for the population \citep{fecher2014open}.
In this paper, we argue that availability can not be fully achieved without accessibility (Box 1).
We identify factors that have particularly affected accessibility in
the natural sciences, and ways scientific developers can address them to ensure reproducible
and automatic scientific workflows. This principles can be generalized and integrated
into the reproducibility curriculum to hopefully provide present learners and future researchers with
the tools for successful scientific reproducibility.
% full workflow automatization and reproducibility in the natural sciences


\bigskip
\bigskip

\noindent\fbox{%
    \parbox{\textwidth}{%
    \textbf{Box 1. The role of accessibility to achieve full availability.}
      Availability is generally defined as the ability or potential of something
      to be used and accessed \citep{availability2022cambridge}.
      Following this definition, accessibility is generally considered a synonym of availability.
      Yet, in practice, availability does not imply accessibility.
      One example we find really useful to practically differentiate the concepts of availability and accessibility
      is the ``marshmallows in an office`` story, which goes like this.
      A couple of brand new marshmallow bags remained in a common office area after
      a social gathering.
      The marshmallows were not claimed by anybody, so they were left in the common are,
      freely available to be taken or eaten by anyone in the office.
      Yet, the marshmallow bags stayed in the common area, unopened for days and even weeks.
      It was not until someone decided to open the bags and placed the marshmallows on a tray,
      that people started actually eating them. They were gone in a matter of hours.
      Scientific products are not sweet as marshmallows, but they are a coveted resource.
      While sharing, describing and documenting makes scientific products available,
      it does not necessarily makes them accessible. For example, documentation of an
      available scientific workflow might be written using expert language that is too foreign
      for a general audience to successfully reproduce the workflow. While some individuals might
      be able to invest the time to learn the expert language to successfully reproduce
      the workflow, this will not be the case for the majority, even if it is
      something that would be useful long term, the short term time investment
      is often too high for individuals to engage with it, making the workflow and its results
      effectively inaccessible and hence unavailable to the majority of the community.
    }%
}
\bigskip


To elaborate on our thesis, we use a widely used scientific workflow in phylogenetics,
a research field within evolutionary biology that focuses
on understanding the shared ancestry of living organisms through the reconstruction of
phylogenetic trees from biological data. Phylogenies provide the
basis to study and understand all biological processes in an evolutionary context
\citep{dobzhansky1973nothing}.
Given the importance of phylogenies, improving reproducibility rates
in phylogenetic research is increasingly relevant for many aspects of the natural sciences.
The Open Tree of Life project (OpenTree) has developed a computational platform that provides
availability of results from phylogenetic research, by creating a standardized database of
phylogenetic data that is used to synthesize a single phylogenetic tree encompassing
all life \citep{opentreeoflife2019synth}.
The OpenTree database Phylesystem \citep{mctavish2015phylesystem} is programmatically
available free of cost to users through OpenTree's Application Programming Interface
(API) services \citep{opentreeAPIs}.
Additionally, software packages written with open source, free of cost programming
languages commonly used in scientific research such as R and Python
\citep{baker2017scientific} have been developed as wrappers that have increased the availability
of OpenTree's API services, by making the OpenTree API functionalities more accessible
to a wider user audience \citep{michonneau2016rotl, mctavish2021opentree}.
Yet, the R and Python OpenTree API wrapper software packages have been mainly used
by computer-literate individuals
to seamlessly establish reproducible workflows to use and reuse expert phylogenetic
knowledge for biological research \citep{sanchez2019datelife, sanchez2021physcraper}
and education \citep{nguyen2020phylotastic, phylotasticedtools, galacticedtools}.
In the 5 years since its release, the R package wrapper for OpenTree \texttt{rotl}
has been cited by more than 190 biological research papers addressing various
topics \citep{googlerotl}.
Comparatively, other R packages for phylogenetics, such as \texttt{GGTREE}
and \texttt{mixOmics}, which were released one year after \texttt{rotl} have been
cited over a thousand times, which might indicate that \texttt{rotl} has not been
adopted as widely.
%% We need a proper statistical distribution of citation numbers per year, to see if rotl is an outlier or not.

While learners in the natural sciences have been engaging independently with R
and Python programming language
as they represent two of the most widely used programming languages in the sciences
% for their fast and seamless statistics and data analysis environment
\citep{baker2017scientific}, computer programming is not traditionally
a core skill formally taught to biologists and naturalists
\citep{sayres2018bioinformatics, wright2019the, williams2019barriers},
However, the OpenTree project demonstrates that efforts to improve availability and reproducibility
in phylogenetics have increased required baseline computational knowledge and skills
in the field.
As computers continue to play a larger role in most scientific disciplines \citep{piccolo2016tools},
baseline computational skill requirements are also increasing across all natural sciences.
% "Thus efforts to increase scientific reproducibility should consider the ubiquity of computers in research."
% In order for reproducible computational tools to be adopted for research, they
% need to be readily accessible for present and future researchers.
Thus, efforts to increase reproducibility rates in the natural sciences must consider
the particularities of scientific workflows that rely on usage of programming languages, such as
availability and accessibility of data and code \citep{peng2011reproducible, sandve2013ten, powers2019open}.
% However, the utility of data resources is limited by the technical challenges of accessing the data.
% In order to motivate reproducible research, that gap needs to be bridged.
% In this work we focus on improving accessibility of code examples and documentation.

We chose a phylogenetics workflow that relies on data and code from OpenTree to
identify the specific barriers to accessibility and reproducibility it presents.
Then, we design ways to overcome or diminish these barriers and apply them to
a series of tutorials and vignettes developed for the OpenTree project.
Finally, we generalize our finding to suggest a set of
principles that can be used as guide to develop code and documentation materials
with increased accessibility that can improve availability and contribute to reproducibility.
Notably, these principles can be incorporated as learning goals into any syllabus for a
course or workshop on best practices for scientific reproducibility.
% In particular, we identify the necessity for documentation that is written down
% using language that is common to the target audience to facilitate examination,
% application, and adoption of code by the wider audience.

\section*{Identifying hurdles to accessibility}
\label{sec:identifying}

Good primary documentation for code is thorough. It describes general usage of individual functions,
the components and variables a function can take, and it should be accompanied with
function usage examples on how to apply it \citep{karimzadeh2018top}.
As opposed to code, primary documentation is written in natural language (i.e.,
any known human language, e.g., English, Spanish, Chinese).
Primary documentation is viewed as a key element for success of a piece of code
\citep{karimzadeh2018top},
which might be why it is also usually written using highly specialized computational jargon (i.e., computationally specific concepts,
words, and phrases) as well as formal scientific language.
While this might be important for formal acceptance of the code by the scientific
and academic community, it often slows down or even
obstructs examination, application, and adoption of code by the general audience \citep{ball2017its}.

Vignettes and tutorials work as secondary pieces of documentation, that help to demonstrate
additional cases of individual function usage, and showcase function associations
that work for specific analysis workflows in more detail.
As secondary documentation has become more common practice and is more flexible
in its form and content, it constitutes an ideal canvas to develop, implement and
test principles that can overcome current barriers to code accessibility.
The tutorials are available at https://mctavishlab.github.io/R\_OpenTree\_tutorials/.

\section*{Addressing hurdles to accessibility: some principles}
\label{sec:addressing}

\subsubsection*{a. Literate programming: Demonstrate code usage with integrative examples}

%% I think I have three different points here:
%% internalizing knowledge is made easier with active learning practices: learn by doing!
%% literate programming allows learning by doing AND at the same time provides a story that connects the acquired knowledge to a real worl problem.
%% By definition, all languages are abstract. We learn them because we grow up with them and they have a direct application for our lives.
%% Programming languages are no different, however, we start learning those later in life and we do not use them as regularly, making it harder for us to learn them and use them.
Pedagogical research shows that active learning practices are one the most effective
ways to take on abstract subjects \citep{freeman2014active}.
Programming computer languages are quite abstract and learning them can be greatly
enhanced by applying an active learning strategy such as a linking its usage to
a ``real world'' or ``human'' application \citep{felder2009active}.

A story-like narrative that links pieces of code together and invites learners
to try the code can greatly support learners to remember what they are doing and
why they are doing it.
This ``literate programming'' paradigm \citep{knuth1984literate, fritzson2002mathmodelica}
makes code more approachable, as it integrates narratives with computer code in
the same document, supporting learners in actively following, remembering and understanding
the code usage \citep{piccolo2016tools}.
Documents developed with ``literate programming'' can be made more accessible by
choosing narratives that are relatable to a more general audience.

We examined available primary documentation for the package \texttt{rotl},
and designed a narrative that required the usage of as many functions as possible.
We demonstrate code applications that are commonly requested by OpenTree users,
but that are not demonstrated in the R package primary documentation.
By framing the function workflow using highly requested uses, the documentation acquires a
narrative arc that is easier to follow and remember by users. This can also facilitate the application
of code to other use cases in biology of interest for the learners.

%% Maybe Figure!! For the tutorial demonstrated here, we used the commonly requested use case of obtaining
%% a phylogenetic tree for all lineages within a specific taxonomic rank.

\subsubsection*{b. Demonstrate errors and warnings thoroughly}

A practice that has become more and more widespread in programming languages pedagogical practices
is the use of typos and mistakes to normalize them for learners and show them how
to solve them when they are outside the classroom \citep{shannon2015live}.
Yet, this is rarely done for written pedagogical materials.
Primary documentation focuses on demonstrating usage function with examples that
work seamlessly, without errors. We argue that the opposite is needed to support
adoption of reproducible workflows and support long term independence in learners
\citep{gaspar2007restoring}.
We demonstrate examples that do not work
as expected and exemplify ways to address them (Figure \ref{fig:first}).

We identify inputs that would give
a wide range of warnings and errors, focusing on demonstrating these cases. This
helps users to not be afraid of errors and warnings, but instead to use them to
their advantage.
We also identify effects of warnings and errors downstream of the workflow.

We identify ways to evaluate inputs to know if they will produce an error, and design
alternatives on what to do when faced with an error or warning, and demonstrate
these alternatives.
One of the most essential skills in programming is interpreting and moving forward
from errors.
Many finely honed tutorials do not trigger errors, which precludes helping students
to develop the tools to understand and address errors when they do encounter them,
as they inevitably will.
On our tutorial, we focus on explaining the meaning and downstream of warnings and errors, and
 showcase ways to detect them before they are triggered (i.e., before using an input
  that would elicit a warning or error). This has two pedagogical benefits:
1) it provides users/students with the means to troubleshoot their own warnings and errors, and
2) it allows users/students to understand with more depth what the function is doing.

% We designed ways to access the different elements of the outputs.

\begin{figure}
\begin{center}
\includegraphics[width=3in]{fig1.png}
\end{center}
\caption{Snapshot of a section of the tutorial website, where we demonstrate a common error. \label{fig:first}}
\end{figure}

\subsubsection*{c. Avoid jargon and expert language}

Besides avoiding formal language, and incorporating elements of pop culture, such as picture
character icons known as ``emojis``, to make the language more familiar to a
broader target audience (see Figure \ref{fig:first}), we made an effort to specifically
complement the primary documentation by identifying
computational concepts that were assumed or were not explained in depth.
We vetted the tutorials with an audience on workshops as well as individual user.
We choose examples that are charismatic for the audience.
For example, when we presented the tutorial for a team specialized in Amphibians,
we tailored the examples using frogs and their allies.


\subsubsection*{d. Make it stable through time}

We published the tutorials on a public, free license, free of cost, and free for
use and reuse repository and persistent website \citep{RopentreeTutorials, RopentreeTutorialsWebsite}.
The tutorial is available for the users to go back to it any time they need it,
and to be passed on to other users (Figure \ref{fig:second}).

\begin{figure}
\begin{center}
\includegraphics[width=3in]{fig2.png}
\end{center}
\caption{Snapshot of the home to our tutorial website, showing part of the schedule. \label{fig:second}}
\end{figure}

Following the carpentries, we created a main version of the tutorial that
is updated. Versions presented on workshops are a copy from the original repository,
and represent a stable and temporal snapshot of the functions and workflows presented
in the tutorial.

%%% \subsubsection*{5. The (now) classics of computational reproducibility}
%%%
%%% Provide all information on package version and system capabilities.
%%%
%%% > [name=Emily Jane McTavish]link to where/and how you did that

\section*{Conclusion}
\label{sec:conclusion}

Ultimately, the long term improvement of reproducibility rates in science will depend
on our ability to intentionally integrate the subject of reproducibility into the
undergraduate curriculum, so college learners and future researchers have the
basis to develop the fundamental skills needed to successfully create reproducible
scientific workflows and materials.

Some universities have been incorporating the subject in their classes (see
\cite{uwlibraries2022, nigms2022}).
The focus of this resources has been for students to develop skills to document their work.
The principles identified and outlined here can be used to set learning goals and
outcomes on new reproducibility syllabus.





We have received emails from senior researchers thanking us for this materials,
 and students have been able to engage using the packages with less hep from the PIs.

The principles to create tutorials described here facilitate adoption of software
 and analysis workflows among researchers at different academic levels, from undergrads
  to established researchers.
It will also help closing the gap between students that had access to computational
 resources (and computational training) from an early age and students that did not.
  Late access to computational resources and training can occur due to lack of
   economic resources, often occurring in households from underrepresented communities
    and minorities.
It can also be due to gender-biased parental and community pressures,
 in which male individuals are more often encouraged to perform activities related to computers,
  while female individuals are discouraged.
% How to balance software acceptance VS. adoption?
These principles can be used to aide not only in improving reproducibility practices,
 but also software adoption in the natural sciences.
%% Discuss: why address accessibility and not other aspects of reproducibility?


Making accessible reproducible workflows has several advantages:
save explanation/training time when analyses are run again by students and collaborators.
save research time for yourself when analyses are run again with more data, a different dataset, a different organism or biological model.
scientific efforts can build off of each other

\bigskip
\begin{center}
{\large\bf SUPPLEMENTARY MATERIAL}
\end{center}

\begin{description}

\item[Title:] Website and GitHub repository containing the complete teaching materials developed and demonstrated here.

\item[GitHub repository link:] \url{https://github.com/McTavishLab/R_OpenTree_tutorials}

\item[Website link:] \url{https://mctavishlab.github.io/R_OpenTree_tutorials}

\end{description}

\bibliography{Manuscript-bibliography}

\end{document}
